src/main/java/com/enderio/base/common/handler/GrindingHandler.java

```diff
@@ -3,53 +3,58 @@
 import com.enderio.EnderIO;
 import com.enderio.base.common.init.EIOItems;
 import net.minecraft.util.RandomSource;
+import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
-import net.minecraftforge.event.GrindstoneEvent;
+import net.minecraft.world.level.block.Blocks;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraftforge.event.entity.player.PlayerInteractEvent;
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 import net.minecraftforge.fml.common.Mod;
 
 @Mod.EventBusSubscriber(modid = EnderIO.MODID)
 public class GrindingHandler {
     @SubscribeEvent
-    public static void onGrindstonePlace(GrindstoneEvent.OnPlaceItem event) {
-        if ((event.getTopItem().is(Items.DEEPSLATE) || event.getTopItem().is(Items.COBBLED_DEEPSLATE)) && event.getBottomItem().is(Items.FLINT)) {
-            event.setOutput(new ItemStack(EIOItems.GRAINS_OF_INFINITY.get(), 1));
-            event.setXp(0);
-        } else if (event.getTopItem().is(Items.COAL) && event.getTopItem().getCount() >= 3 && event.getBottomItem().is(Items.FLINT)) {
-            event.setOutput(new ItemStack(EIOItems.POWDERED_COAL.get(), 1));
-            event.setXp(0);
-        }
-    }
-
-    @SubscribeEvent
-    public static void onGrindstoneResult(GrindstoneEvent.OnTakeItem event) {
-        if ((event.getTopItem().is(Items.DEEPSLATE) || event.getTopItem().is(Items.COBBLED_DEEPSLATE)) && event.getBottomItem().is(Items.FLINT)) {
-            ItemStack top = event.getTopItem().copy();
-            ItemStack bottom = event.getBottomItem().copy();
-
-            top.shrink(1);
-
-            // 33% chance to destroy the flint
-            if (RandomSource.create().nextInt(3) == 0) {
-                bottom.shrink(1);
+    public static void onRightClickBlock(PlayerInteractEvent.RightClickBlock event) {
+        //Server Side Only
+        if(!event.getSide().isServer())
+            return;
+        Player player = event.getEntity();
+        if (!player.isCrouching())
+            return;
+        BlockState target = event.getLevel().getBlockState(event.getPos());
+        if(!(target.is(Blocks.GRINDSTONE) || target.is(Blocks.OBSIDIAN) || target.is(Blocks.CRYING_OBSIDIAN)))
```

Comment:
Replace the obsidian check with the obsidian Tag check. The TagKey can be Found in Tags.Blocks.OBSIDIAN

Reply:
C++ Coder habits again 😛 
Let me know if what I wrote in the changelog is too wordy

Reply:
that one was close, crying obsidian is not in the obsidian tag, so that check needs to persist